game/quake2_md2.ksy

```diff
@@ -0,0 +1,227 @@
+meta:
+  id: quake2_md2
+  title: Quake II player model (version 8)
+  application: Quake II
+  file-extension: md2
+  license: CC0-1.0
+  endian: le
+doc: |
+  The MD2 format is used for 3D animated models in id Sofware's Quake II.
+
+  A model consists of named `frames`, each with the same number of `vertices`
+  (`vertices_per_frame`). Each such vertex has a `position` in model space,
+  and a `normal_index` which you must look up to get its normal. Each vertex
+  has the same topological "meaning" across frames, in terms of triangle and
+  texture info; it just varies in position and normal for animation purposes.
+
+  How the vertices form triangles is defined via disjoint `triangles` or via
+  `gl_cmds` (which allows strip and fan topology.) Each triangle contains three
+  `vertex_indices` into frame vertices, and three `tex_point_indices` into
+  global `tex_coords`. Each texture point has pixel coords `u_px` and `v_px`
+  ranging from 0 to `skin_{width,height}_px` respectively, along with
+  `{u,v}_normalized` ranging from 0 to 1 for your convenience.
+
+  A GL command has a `primitive` type (`TRIANGLE_FAN` or `TRIANGLE_STRIP`) along
+  with some `vertices`. Each GL vertex contains `tex_coords_normalized` from 0
+  to 1, and a `vertex_index` into frame vertices.
+
+  A model may also contain `skins`, which are just file paths to PCX images.
+  However, this is empty for many models, in which case it is up to the client
+  (e.g. Q2PRO) to offer skins some other way (e.g. by similar filename in the
+  current directory.)
+
+  There are 198 `frames` in total, partitioned into a fixed set of ranges used
+  for different animations. Each frame has a standard `name` for humans, but the
+  client just uses their index and the name can be arbitrary. The frames are:
+
+   INDEX     NAME SUFFIX NOTES
```

Comment:
I added this section, formatted the table like this from the POV of someone reading a text file. However, now I think about it, will the doc section be rendered as Markdown in a web page? In which case, I should make this a Markdown table?

Reply:
I've made this a Markdown table so it should display correctly. I think this PR is ready to merge.

Reply:
@jdjakub:

> will the doc section be rendered as Markdown in a web page?

Yes. The top-level `doc` key is rendered at https://formats.kaitai.io/ as the format description.

> I've made this a Markdown table so it should display correctly.

Unfortunately, the Markdown dialect we use is [CommonMark](https://commonmark.org/), which doesn't provide markup for tables. So you could use either use HTML syntax (`<table>` etc. - it would probably lose the readability a bit), or you may insert a plaintext code block `` ``` `` with an ASCII-formatted table.

ASCII table is fine, if you ask me.

Reply:
I've now expanded the table as a preformatted block. Do you have any thoughts on my Sep 21 comment below beginning "I have added this machine-readable version of part of the animation table..."?